src/Ilios/CoreBundle/Controller/UserController.php

```diff
@@ -193,25 +193,49 @@ public function postAction(Request $request)
     {
         try {
             $handler = $this->getUserHandler();
-            $data = $this->getPostData($request);
-
-            if (empty($data['icsFeedKey'])) {
-                //create an icsFeedKey for the new user
-                $random = random_bytes(128);
-                $key = microtime() . '_' . $random;
-                $data['icsFeedKey'] = hash('sha256', $key);
+            $arr = $this->getPostData($request);
+            $userIds = [];
+            $unsavedUsers = [];
+            $count = 0;
+            foreach ($arr as $data) {
+                if (empty($data['icsFeedKey'])) {
+                    //create an icsFeedKey for the new user
+                    $random = random_bytes(128);
+                    $key = microtime() . '_' . $random;
+                    $data['icsFeedKey'] = hash('sha256', $key);
+                }
+
+                $user = $handler->post($data);
+
+                $authChecker = $this->get('security.authorization_checker');
+                if (! $authChecker->isGranted('create', $user)) {
+                    throw $this->createAccessDeniedException('Unauthorized access!');
```

Comment:
having this fail after the 50 users mark will cause the request to fail, with part of the users already added to the system.
suggesting pulling this out into a separate loop that runs prior to insert loop.